cdap-api/src/main/java/co/cask/cdap/api/lineage/Destination.java

```diff
@@ -0,0 +1,119 @@
+/*
+ * Copyright © 2018 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package co.cask.cdap.api.lineage;
+
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Objects;
+import javax.annotation.Nullable;
+
+/**
+ * Represents the information about the destination dataset.
+ */
+public class Destination {
+  private final String name;
+  private final String namespace;
+  private final Map<String, String> properties;
+
+  private Destination(String name, @Nullable String namespace, Map<String, String> properties) {
+    this.name = name;
+    this.namespace = namespace;
+    this.properties = Collections.unmodifiableMap(new HashMap<>(properties));
+  }
+
+  /**
+   * @return the name of the destination
+   */
+  public String getName() {
+    return name;
+  }
+
+  /**
+   * @return the namespace name if the destination exist in different namespace,
+   * otherwise {@code null} is returned
+   */
+  public String getNamespace() {
+    return namespace;
+  }
+
+  /**
+   * @return the properties associated with the destination for lineage purpose
+   */
+  public Map<String, String> getProperties() {
+    return properties;
+  }
+
+  /**
+   * Return the destination as defined by the provided dataset name.
+   * @param datasetName the name of the dataset
+   * @return the destination
+   */
+  public static Destination ofDataset(String datasetName) {
```

Comment:
I assume this call means the dataset is in the same namespace? By how do we represent entities that are not namespaced? (e.g. a DB sink).

Reply:
Yes thats right. This call is for the dataset which is in the same namespace. For DB sink we still create the dataset internally for the lineage purpose (as indicated by the reference name config). So App will call this API with that name which is specified by the reference name for the DB Sink.